File: app/src/main/java/ua/fvadevand/reminderstatusbar/adapters/IconAdapter.java
package ua.fvadevand.reminderstatusbar.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.DrawableRes;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import ua.fvadevand.reminderstatusbar.R;

public class IconAdapter extends RecyclerView.Adapter<IconAdapter.IconViewHolder> {

    private List<Integer> mIconIds;
    private OnIconClickListener mListener;

    public IconAdapter(List<Integer> iconIds, OnIconClickListener listener) {
        mIconIds = iconIds;
        mListener = listener;
    }

    @NonNull
    @Override
    public IconViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.icon_list_item, parent, false);
        return new IconViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull IconViewHolder holder, int position) {
        holder.mIconView.setImageResource(mIconIds.get(position));
    }

    @Override
    public int getItemCount() {
        return mIconIds == null ? 0 : mIconIds.size();
    }

    public interface OnIconClickListener {
        void onIconClick(@DrawableRes int iconId);
    }

    class IconViewHolder extends RecyclerView.ViewHolder {

        ImageView mIconView;

        IconViewHolder(@NonNull View itemView) {
            super(itemView);
            mIconView = itemView.findViewById(R.id.iv_icon);
            itemView.setOnClickListener(v -> {
                int position = getAdapterPosition();
                if (mListener != null && position != RecyclerView.NO_POSITION) {
                    mListener.onIconClick(mIconIds.get(position));
                }
            });
        }
    }
}

File: app/src/main/java/ua/fvadevand/reminderstatusbar/adapters/IconAdapter.kt
package ua.fvadevand.reminderstatusbar.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.annotation.DrawableRes
import androidx.recyclerview.widget.RecyclerView
import ua.fvadevand.reminderstatusbar.databinding.ListItemIconBinding

class IconAdapter(
    private val iconIds: List<Int>,
    private val onClickListener: (iconId: Int) -> Unit
) : RecyclerView.Adapter<IconAdapter.IconViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): IconViewHolder {
        val itemBinding = ListItemIconBinding.inflate(
            LayoutInflater.from(parent.context),
            parent,
            false
        )
        return IconViewHolder(itemBinding)
    }

    override fun onBindViewHolder(holder: IconViewHolder, position: Int) {
        holder.bind(iconIds[position])
    }

    override fun getItemCount(): Int {
        return iconIds.size
    }

    inner class IconViewHolder(
        private val itemBinding: ListItemIconBinding
    ) : RecyclerView.ViewHolder(itemBinding.root) {

        init {
            itemBinding.root.setOnClickListener {
                val position = adapterPosition
                if (position in 0 until itemCount) {
                    onClickListener(iconIds[position])
                }
            }
        }

        fun bind(@DrawableRes iconId: Int) {
            itemBinding.ivIcon.setImageResource(iconId)
        }
    }
}
